gluster/cli/peer.py

```diff
@@ -47,6 +47,34 @@ def detach(host):
     cmd = ["detach", host]
     return peer_execute(cmd)
 
+def detachAll():
+    """
+    Removes All Hosts from Cluster
+
+    :returns: success and failure messages, raises
+     GlusterCmdException((rc, out, err)) on error
+    """
+    try:
+        peers = parse_peer_status(peer_execute_xml(["status"]))
+    except:
+        return parse_peer_status(peer_execute_xml(["status"]))
+    if len(peers) == 0:
+        return "No peers to Detach"
+    else:
+        print "Found "+str(len(peers))+" peers"
+        for peer in peers:
+                host = peer["hostname"]
+                if peer["connected"] == "Connected":
+                    cmd = ["detach",host]
+                    try:
+                        result = peer_execute(cmd)
+                        out = str(host)+" "+result
+                        print out
+                    except:
+                        return peer_execute(cmd)
+                else:
+                    return str(host)+" is not connected. Can't Detach"
+    return "ALL PEERS DETACHED SUCCESSFULLY"
```

Comment:
Please don't put the caller through the nasty business of processing these strings. Python is simple: if there is no consumable value to be returned and all the caller needs to know is success or failure, then return nothing on success, raise exception on failure.